raster/src/main/scala/geotrellis/raster/io/geotiff/Float64GeoTiffSegmentCollection.scala

```diff
@@ -16,22 +16,18 @@
 
 package geotrellis.raster.io.geotiff
 
-import geotrellis.raster._
-import geotrellis.raster.io.geotiff.compression._
-import spire.syntax.cfor._
-
 trait Float64GeoTiffSegmentCollection extends GeoTiffSegmentCollection {
   type T = Float64GeoTiffSegment
 
   val bandType = Float64BandType
-  val noDataValue: Option[Double]
+  def noDataValue: Option[Double]
 
-  lazy val createSegment: Int => Float64GeoTiffSegment = noDataValue match {
+  lazy val decompressGeoTiffSegment = noDataValue match {
     case None =>
-      { i: Int => new Float64RawGeoTiffSegment(getDecompressedBytes(i)) }
-    case Some(nd) if (nd == Float.NaN) =>
-      { i: Int => new Float64ConstantNoDataGeoTiffSegment(getDecompressedBytes(i)) }
+      (i: Int, bytes: Array[Byte]) => new Float64RawGeoTiffSegment(decompressor.decompress(bytes, i))
+    case Some(nd) if nd.isNaN =>
```

Comment:
We always use `java.lang.Float.isNaN(nd)`, because for some reason it's significantly faster than `.isNaN`